app/commands/user_command.rb

```diff
@@ -0,0 +1,84 @@
+class UserCommand
+  attr_reader :user
+  def initialize(user)
+    @user = user
+  end
+
+  def soft_delete_with_relations
+    efficiently_delete_relations
+    user.soft_delete
+  end
+
+  def restore_with_relations
+    efficiently_restore_relations
+    user.restore
+  end
+
+  def destroy_with_relations
+    efficiently_destroy_relations
```

Comment:
I know you questioned having a method that only points to another method - but this one hides the actual implementation to private method. OK to leave it as is?